Pilot/pilotCommands.py

```diff
@@ -487,19 +482,29 @@ def execute(self):
       self.log.error("Wrong Command output %s" % result)
       sys.exit(1)
 
-    cfg = []
     # If NumberOfProcessors or MaxRAM are defined in the resource configuration, these
     # values are preferred
     if "WholeNode" in self.pp.tags:
-      numberOfProcessors = numberOfProcessorsOnWN
+      self.pp.payloadProcessors = numberOfProcessorsOnWN
     if self.pp.maxNumberOfProcessors > 0:
-      numberOfProcessors = min(numberOfProcessorsOnWN, self.pp.maxNumberOfProcessors)
+      self.pp.payloadProcessors = min(numberOfProcessorsOnWN, self.pp.maxNumberOfProcessors)
 
-    if not numberOfProcessors:
+    if not self.pp.payloadProcessors:
```

Comment:
I think this can never happen, cause you explicitely set this variable in 475

Reply:
but it does no harm

Reply:
I think it can happen if dirac-wms-get-wn-parameters returns zero processors. (It is the same as the current code anyway.)